Feature/salesforce integration

maxpearl opened the pull request:
Addresses #2054

Adds an environment variable: SALESFORCE_INTEGRATION as well as a new optional settings file.

frjo commented on hypha/settings/dev.py:
```diff
@@ -145,3 +145,11 @@
+        from .salesforce import *
+    except ImportError:
+        pass
+    DATABASES.update(SALESFORCE_DB)
```
Since this needs to be on all environments I thinks it better to put it in `base.py`. Then we can add it only once.

frjo commented on hypha/settings/salesforce.py.example:
```diff
@@ -0,0 +1,14 @@
+    'USER': 'user@email.com',
+    'PASSWORD': 'Password and security token concatenated',
+    'HOST': 'https://login.salesforce.com' # Or test or URL of salesforce org
+}
```
I think we should make all these environment variables. On hosting environments like Heroku you can't upload a separate configuration file but setting environment variables is easy and reasonable secure.

frjo replied:
Having the option to use environment variables or a settings file might be the absolut best.

maxpearl commented:
I added a new commit which moves all of the variables to the environment. I added a sf.env file to docker-compose for sensitive data, and added that to .gitignore.

frjo commented on hypha/settings/base.py:
```diff
@@ -201,12 +203,34 @@
+            default=f"postgres:///{APP_NAME}"
+        )
+    }
 
```
I would prefer to put all the Salesforce settings in a separate section. Something like this.

~~~~
# Salesforce integration

if env.get('SALESFORCE_INTEGRATION', 'false').lower().strip() == 'true':
    DATABASES = {
        **DATABASES,
        'salesforce': {
            'ENGINE': 'salesforce.backend',
            'CONSUMER_KEY': env.get('SALESFORCE_CONSUMER_KEY', ''),
            'CONSUMER_SECRET': env.get('SALESFORCE_CONSUMER_SECRET', ''),
            'USER': env.get('SALESFORCE_USER', ''),
            'PASSWORD': env.get('SALESFORCE_PASSWORD', ''),
            'HOST': env.get('SALESFORCE_LOGIN_URL', '')
        }
    }

    SALESFORCE_QUERY_TIMEOUT = (30, 30)  # (connect timeout, data timeout)

    DATABASE_ROUTERS = [
        "salesforce.router.ModelRouter"
    ]
~~~~

frjo commented on .gitignore:
```diff
@@ -12,6 +12,7 @@
 node_modules/
 /npm-debug.log
 /.idea/
+sf.env
```
Maybe name this "salesforce.env" just to be explicit.

maxpearl replied:
Good idea.

frjo commented:
@maxpearl You have write access to this repo so you do not need to keep things in your own fork. Feel free to commit directly to the Hypha salesforce-integration branch.

maxpearl commented:
Thanks, but I'd rather use this workflow so I can get comments on my code before I add them to the main repo, if that's OK with you. But I'm happy to merge the pull requests after we decide the code is going to work.
